src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
Ah OK I like if it's not our intention to leak internal state back to the user, and understand that if there's current reliance on it, we can't just change it now. So I reverted my simplification.

The technical difficulty is that this object, on which it attempts to set `cmin`/`cmax` may *not* exist: `marker` inherits props to `marker.line` but `marker.line` might not have been specified among the input. Therefore the code attempts to set `cmin`/`cmax` on `undefined`. In this case I'll need to create a new object for it. You were right when you said coercions etc. are among the harder to tackle parts. My initial way of creating the object doesn't look pretty so I might need to muck around Lib.nestedProperty
```
    if(containerStr) {
        container = Lib.nestedProperty(trace, containerStr).get();
        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
        if(!inputContainer) {
            Lib.nestedProperty(trace._input, containerStr).set({fixme:1}); // empty array won't work
            inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
        }
    } else {
        container = trace;
        inputContainer = trace._input;
    }
```